sqlite/comdb2build.c

```diff
@@ -1675,3 +1673,2269 @@ void comdb2WriteTransaction(Parse *pParse)
+        if (key == 0) {
+            setError(pParse, SQLITE_ERROR, "FK: Local key used in the foreign "
+                                           "key constraint could not be "
+                                           "found.");
+            goto cleanup;
+        }
+
+        /* Locate the parent key. */
+        for (int j = 0; j < table->constraints->nrules; j++) {
+            parent_key = 0;
+            parent_table = getdbbyname(table->constraints[i].table[j]);
+            if (parent_table == 0) {
+                pParse->rc = SQLITE_ERROR;
+                sqlite3ErrorMsg(pParse, "FK: Parent table '%s' not found.",
+                                table->constraints[i].table[j]);
+                goto cleanup;
+            }
+
+            for (int k = 0; k < parent_table->schema->nix; k++) {
+                if (strcasecmp(parent_table->schema->ix[k]->csctag,
+                               table->constraints[i].keynm[j]) == 0) {
+                    parent_key = parent_table->schema->ix[k];
+                }
+            }
+            if (parent_key == 0) {
+                setError(pParse, SQLITE_ERROR, "FK: Referenced key used in the "
+                                               "foreign key constraint could "
+                                               "not be found.");
+                goto cleanup;
+            }
+
+            assert(key->nmembers == parent_key->nmembers);
+
+            constraint = comdb2_calloc(ctx->mem, 1, sizeof(struct constraint));
+            if (constraint == 0) goto oom;
+            centry =
+                comdb2_calloc(ctx->mem, 1, sizeof(struct comdb2_constraint));
+            if (centry == 0) goto oom;
+
+            constraint->referenced_table =
+                comdb2_strdup(ctx->mem, parent_table->dbname);
+            if (constraint->referenced_table == 0) goto oom;
+            constraint->ncols = parent_key->nmembers;
+            constraint->flags = table->constraints[i].flags;
+
+            for (int p = 0; p < constraint->ncols; p++) {
+                constraint->column[p] =
+                    comdb2_strdup(ctx->mem, key->member[p].name);
+                if (constraint->column[p] == 0) goto oom;
+                constraint->referenced_column[p] =
+                    comdb2_strdup(ctx->mem, parent_key->member[p].name);
+                if (constraint->referenced_column[p] == 0) goto oom;
+            }
+
+            centry->constraint = constraint;
+            listc_abl(&ctx->constraint_list, centry);
+        }
+    }
+    return 0;
+
+oom:
+    setError(pParse, SQLITE_NOMEM, "System out of memory");
+
+cleanup:
+    return 1;
+}
+
+void comdb2AlterTableStart(
+    Parse *pParse, /* Parser context */
+    Token *pName1, /* First part of the name of the table. */
+    Token *pName2, /* Second part of the name of the table. */
+    int dryrun     /* Whether its a dryrun? */
+    )
+{
+    struct comdb2_ddl_context *ctx;
+
+    if (isRemote(pName1, pName2)) {
```

Comment:
"main" is special, it is always the local db.  Also someone could use the dbname.tblname, where dbname is the local db name.